dask/dataframe/multi.py

```diff
@@ -232,9 +232,16 @@ def require(divisions, parts, required=None):
 }
 allowed_left = ("inner", "left", "leftsemi", "leftanti")
 allowed_right = ("inner", "right")
+merge_chunk_dispatch = Dispatch("merge_chunk")
 
 
 def merge_chunk(lhs, *args, **kwargs):
+    func = merge_chunk_dispatch.dispatch(type(lhs))
+    return func(lhs, *args, **kwargs)
+
+
+@merge_chunk_dispatch.register((pd.DataFrame, pd.Series, pd.Index, pd.Categorical))
+def merge_chunk_pandas(lhs, *args, **kwargs):
```

Comment:
Instead of dispatching at the `merge_chunk` level I'm wondering if we should implement `union_categoricals` in cuDF and dispatch on that instead?

Reply:
We could do that but that would need `cudf.Categorical` to be implemented.

Reply:
We could choose not to implement `union_categoricals` quite yet but dispatch on an internal function for now that returns a `CategoricalIndex` for example.

Reply:
Looks like this is the relevant cuDF PR ( https://github.com/rapidsai/cudf/pull/8332 )

Reply:
Introduced only `union_categoricals` dispatch